enhancements/ingress/aws-lb-eip-selection.md

```diff
@@ -0,0 +1,443 @@
+     We will be comparing if the number of public subnets added to the install-config matches with len(eipAllocations).
+     However, The problem is that the AWS CCM can select subnets that aren't provided in the BYO Subnets, see https://issues.redhat.com/browse/OCPBUGS-17432.
+Here's one recommended option:
+
+The Installer should count all LB subnets by predicting what subnets be chosen by the AWS CCM 
+(i.e. any subnet without another cluster's kubernetes.io/cluster/<cluster-id> tag). 
+We can call this Predicted LB Subnet Count.
+
+We can examine the following scenarios:
+
+##### BYO Subnet Count != EIPs Allocations && BYO Subnet Count == Predicted LB Subnet count && Predicted LB Subnet count != EIPs Allocations:
+
+Throw a simple error that just says, EIP != Provided Subnets:
+The number of EIP Allocations does not equal the number of provided Subnets, the cluster will fail.
+
+##### BYO Subnet Count == EIPs Allocations && BYO Subnet Count != Predicted LB Subnet count && Predicted LB Subnet count != EIPs Allocations:
+
+Throw a error that says something to the extend of that the LB will select more subnets and the cluster will fail:
+Or though the number of EIP Allocations equals the number of provided Subnets, it does not equal the number of Subnets that will be selected by
+the LoadBalancer and the cluster will fail, please review the kubernetes.io/cluster/<cluster-id> subnet tags are set correctly.
+
+##### BYO Subnet Count != EIPs Allocations && BYO Subnet Count != Predicted LB Subnet count && Predicted LB Subnet count == EIPs Allocations:
+
+This is an odd scenario. The user got the # of a EIPs == Predicted LB Subnet count, I suppose because they anticipated the AWS CCM's generous 
+selection of subnets. This is valid scenario for no error message.
+
+##### BYO Subnet Count == EIPs Allocations && BYO Subnet Count == Predicted LB Subnet count && Predicted LB Subnet count == EIPs Allocations:
+
+Obvious valid scenario.
+
+I think we need some feedback from installer-team, @patrickdillon or @mtulio or @sadasu on this type of validation. Should the installer team consider
+the Predicted LB Subnets != BYO Subnet Count scenario as not valid? And possibly block future installs as a resolution
+to https://issues.redhat.com/browse/OCPBUGS-17432? That would make EIP Allocation a lot easier, but not sure if that's realistic. 
+
+4. #### Validation to check if EIPs are not already assigned to resources.
+EIPs can be assigned to many resource types, like Nat Gateways, *LBs, Instances, etc. The attribute associationId will be set when the EIP is already associated.
+To mitigate this we could add that validation, at least on installer, to provide quick-feedback (fail when validate install-config) to the user when the provided EIP is already associated to another resource.
+It would be nice to have a validation before setting the annotation to CCM, keeping the operator degraded before disrupting the service.
+
```

Comment:
Could we add a small section on the defaulting mechanics for the IngressController? It's definitely an implementation detail, but an important one that feels important enough to mention here. Something @Miciah  or @JoelSpeed might be interested in commenting on. I wonder if doing status "right" was a bit of a mistake, because now things are becoming inconsistent in the API. Either way, that that ship sailed in 4.17.

```suggestion
#### IngressConfig EIP Allocation Defaulting Mechanics for Ingress Controller

Traditionally, the Ingress Operator has populated default values from the Ingress Config into the `status`, making `status` effectively reflect the desired state of the IngressController. However, since `eipAllocations` in `status` represents the **actual** state, not the **desired** state, the default `eipAllocations` values must be set in the `spec` when the Ingress Operator initially admits the IngressController.

This approach is new. The Ingress Operator does not typically set default values in `spec` for load balancer configurations if the user hasn’t explicitly provided them. While this defaulting pattern is more consistent with Kubernetes conventions for `spec` and `status` (and is also our only option in this situation), it's important to acknowledge that this inconsistency in defaulting behavior could cause confusion for users.
```

Reply:
@gcs278 @miheer if we were to make all the `status` fields of the Ingress Config consistent and populate them with the **actual** state, is the issue that there is no field that reflects the defaults?

Reply:
IngressConfig is no longer relevant.